src/orion/core/io/database/__init__.py

```diff
@@ -255,11 +252,169 @@ class DuplicateKeyError(DatabaseError):
     pass
 
 
-# pylint: disable=too-few-public-methods,abstract-method
-class Database(AbstractDB, metaclass=SingletonFactory):
-    """Class used to inject dependency on a database framework.
+# pylint: disable=too-few-public-methods,abstract-method,too-many-public-methods
+class Database(Wrapper, metaclass=SingletonType):
```

Comment:
I would drop the abstract and make a wrapper like we do for views; automatically fetch wrapped concept's attributes if not overriden by the wrapper.

Reply:
You mean, drop the `ABCMeta` metaclass from `AbstractDB`?